torch/quantization/fake_quantize.py

```diff
@@ -1,34 +1,67 @@
 from __future__ import absolute_import, division, print_function, unicode_literals
 import torch
 from torch.nn import Module
-from .observer import default_observer, _with_args
+from .observer import MinMaxObserver, _with_args
 
 class FakeQuantize(Module):
     ''' Simulate the quantize and dequantize operations in training time.
+    The output of this module is given by
+
+    x_out = (clamp(round(x/scale + zero_point), quant_min, quant_max)-zero_point)*scale
+
+
+
+    * :attr:`scale` defines the scale factor used for quantization.
+
+    * :attr:`zero_point` specifies the quantized value to which 0 in floating point maps to
+
+    * :attr:`quant_min` specifies the minimum allowable quantized value.
+
+    * :attr:`quant_max` specifies the maximum allowable quantized value.
+
+    * :attr:`fake_quant_enable` controls the application of fake quantization on tensors, note that
+      statistics can still be updated.
+
+    * :attr:`observer_enable` controls statistics collection on tensors
+
+    * :attr:`dtype` specifies the quantized dtype that is being emulated with fake-quantization,
+     allowable values are torch.qint8 and torch.quint8. The values of quant_min and quant_max should
+     be chosen to be consistent with the dtype
+
+
+    Args:
+        observer (module): Module for observing statistics on input tensors and calculating
+        scale and zero-point.
+        quant_min (int): The minimum allowable quantized value.
+        quant_max (int): The maximum allowable quantized value.
+        observer_kwargs (optional): Arguments for the observer module
+
+    Attributes:
+        observer (Module): User provided module that collects statistics on the input tensor and
+                           provides a method to calculate scale and zero-point.
+
+    """
     Args:
-        `qconfig`: object that encodes configuration info for quantization
-        `observer_module`: Observer module that records stats of weights and
-        activations
-        `calcqparam`: A function that calculates quantization parameters
+        `observer`: Observer module that records stats of input tensor
+        `quant_min`: Tensors are fake-quantized corresponding to the
+        `quant_max`: A function that calculates quantization parameters
         given the stats
+        `observer_kwargs`
```

Comment:
why do we want this? why not just ask users to apply the observer_kwards with `with_args` on the `observer` argument?

Reply:
Thats because we need to have an observer factory, the kwargs for the observer have to be passed when the observer module is being instantiated.

Reply:
doesn't `with_args` do the same thing?

Reply:
it's a syntactic thing (and was my idea originally). Rationale is that FakeQuant almost "extends" observer. It doesn't prevent doing with_args either, but it's much shorter:

Compare:
```
FakeQuantize.with_args(observer=MinMaxObserver.with_args(dtype=torch.quint8), quant_min=2)
FakeQuantize.with_args(observer=MyCrazyObserver.with_args(dtype=torch.quint8), quant_min=2)
```
with
```
FakeQuantize.with_args(dtype=torch.quint8, quant_min=2)
FakeQuantize.with_args(observer=MyCrazyObserver, dtype=torch.quint8, quant_min=2)
```

Reply:
I see, makes sense

Reply:
Great, was about to write the same thing here: @dzhulgakov : I feel that either API makes sense, and I prefer having explicit with_args for observer. If both of you agree on the current API, we can keep it as is.
@jerryzh168 : Can you stamp this so that I can land this? :-)